handlers/handlers.go

```diff
@@ -194,10 +195,131 @@ func (s *ServiceSet) serviceHTML(db *mbtiles.DB) handlerFunc {
 	}
 }
 
+type tileCoord struct {
+	z    uint8
+	x, y uint64
+}
+
+// extractTileCoord extracts and returns tile coordinates and an optional
+// extension from a byte slice and returns them. In case the byte slice is too
+// short or the number parsing cannot be performed, the returned error is
+// non-nil.  pcs is expected to be in order (z, y, x) when the parameter swapXY
+// is false, and in order (z, x, y) when swapXY is true. The extension is
+// extracted from the third element of pcs. Any elements in pcs beyond the
+// third are ignored.
+func extractTileCoord(pcs []string, swapXY bool) (tc tileCoord, ext string, err error) {
```

Comment:
@fawick I would prefer that we hand in the string components directly, and not worry about flipping them.  Putting them in proper order should be responsibility of caller, which would reduce complexity here.   `parse` might be a more appropriate term here than `extract` but I could be convinced either way. 😄 

`func parseTileCoords(z, x, y string)`

Reply:
On second thought, what if we made parsing a method of `Tile` above?  So that we can call `Tile.FromString(z, x, y)`?  (where z, x, y are strings).